client/ayon_applications/process.py

```diff
@@ -0,0 +1,682 @@
+"""Handling of processes in Ayon Applications."""
+from __future__ import annotations
+import contextlib
+import os
+import platform
+import subprocess
+import json
+import logging
+import sqlite3
+import threading
+from dataclasses import dataclass
+from hashlib import sha256
+from pathlib import Path
+from typing import NamedTuple, Optional
+
+from ayon_core.lib import (
+    Logger,
+    get_launcher_local_dir,
+)
+
+
+class ProcessIdTriplet(NamedTuple):
+    """Triplet of process identification values."""
+    pid: int
+    executable: str
+    start_time: Optional[float]  # the same goes for start time
+
+
+@dataclass
+class ProcessInfo:
+    """Information about a process launched by the addon.
+
+    Attributes:
+        name (str): Name of the process.
+        executable (Path): Path to the executable.
+        args (list[str]): Arguments for the process.
+        env (dict[str, str]): Environment variables for the process.
+        cwd (str): Current working directory for the process.
+        pid (int): Process ID of the launched process.
+        active (bool): Whether the process is currently active.
+        output (Path): Output of the process.
+
+    """
+
+    name: str
+    executable: Path
+    args: list[str]
+    env: dict[str, str]
+    cwd: str
+    pid: Optional[int] = None
+    active: bool = False
+    output: Optional[Path] = None
```

Comment:
Is this the content of stdout/stderr files? Shouldn't that be loaded in a separate way (so you don't have to open the large files to list processes).

Reply:
It is not a content, it is a path. And it is loaded in a separate way in the Manger tool.

Reply:
Then the attribute name is not clear and confusing, should be `output_path`.